sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,164 @@
+            cleanHeaders[key] = regexSubscriptionReplace(dirtyHeaders[key])
+        }
+        if cleanHeaders.isEmpty {
+            cleanRequest.allHTTPHeaderFields = nil
+        } else {
+            cleanRequest.allHTTPHeaderFields = cleanHeaders
+        }
+        
+        cleanRequest.url = URL(string: regexSubscriptionReplace(request.url?.absoluteString))
+        cleanRequest.mainDocumentURL = URL(string: regexSubscriptionReplace(request.mainDocumentURL?.absoluteString))
+        cleanRequest.httpBody = regexSubscriptionReplace(String(data: request.httpBody, encoding: .utf8))?.data(using: .utf8)
+    
+        return cleanRequest
+    }
+    
+    static func regexSubscriptionReplace(_ string: String?)->String?{
+        guard string != nil else {
+            return nil
+        }
+        var editedString = string!
+        var range = NSRange(location: 0, length: editedString.utf8.count)
+        if subscriptionIDRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+            editedString = subscriptionIDRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "$1/99999999-9999-9999-9999-999999999999")
+        }
+        range = NSRange(location: 0, length: editedString.utf8.count)
+        if graphSubscriptionIDRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+            editedString = graphSubscriptionIDRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "$1/99999999-9999-9999-9999-999999999999")
+        }
+        return editedString
+    }
+    
+    static func scrubSubscriptionIDs(response: Foundation.URLResponse, data: Data?) -> URLRequest {
+        fatalError("Unimplemented")
+    }
+    
+    public static func scrubRequests(request: URLRequest) -> URLRequest {
+        var cleanedRequest = request
+        cleanedRequest = scrubSubscriptionIDs(request: cleanedRequest)
+        
+        return cleanedRequest
+    }
+    
+    static func scrubResponse(response: Foundation.URLResponse,data: Data?) -> (Foundation.URLResponse, Data?) {
+        fatalError("Unimplemented")
+    }
+}
+
+
+extension Array where Element == Any {
+    func dictionaryForIndex(_ index: Int) throws -> Dictionary<String,Any> {
+        if let dictionary = self[index] as? Dictionary<String, Any> {
+            return dictionary
+        }
+        throw "Not a dictionary"
+    }
+    
+    func arrayForIndex(_ index: Int) throws -> Array<Any> {
+        if let array = self[index] as? Array<Any> {
+            return array
+        }
+        throw "Not an array"
+    }
+    
+    func stringForIndex(_ index: Int) throws -> String {
+        if let string = self[index] as? String {
+            return string
+        }
+        throw "Not a string"
+    }
+    
+}
+
+ extension Dictionary where Key == String, Value == Any {
+    
+    static func findValue(key: String, dictionary: Dictionary<String,Any>) -> Any? {
```

Comment:
Don't understand why this would be static.

```suggestion
    func value(forKey key: String) -> Any? {
```